op/main.py

```diff
@@ -116,10 +116,19 @@ def _get_event_args(charm, bound_event):
         relation_name = os.environ['JUJU_RELATION']
         relation_id = int(os.environ['JUJU_RELATION_ID'].split(':')[-1])
         relation = model.get_relation(relation_name, relation_id)
-        if issubclass(event_type, op.charm.RelationUnitEvent):
+        remote_app_name = os.environ.get('JUJU_REMOTE_APP')
+        if remote_app_name:
+            remote_app = model.get_app(remote_app_name)
+        else:
+            remote_app = None
+
+        if remote_app:
```

Comment:
I think we have to use 2.7 as a base for the framework because of the features such as application relation data, podspec v2, actions/functions for k8s charms.

As far as I understand from the discussions we had up until this point, we will not support `leader-{get,set}` and instead use peer app relation data.

Hence I assume unconditionally here that `JUJU_REMOTE_APP` is available.

Let's have a discussion about this during our call.

Reply:
> Do we actually have a reason today to limit ourselves to juju 2.7?
> The things mentioned above don't seem relevant for this

Leader data is a major point. Are you suggesting we have one public implementation exposed to the framework users that uses `leader-{get, set}` before 2.7 and peer application relation data afterwards? Besides having to implement things in 2 different ways we would add an upgrade step as well.

Reply:
Per the discussion during our call:

* I will update the PR to obtain a remote application name from a remote unit name to support earlier versions;
* We will not implement support for leader-{get, set} until somebody asks for it for Juju < 2.7;
   * If we implement it, it will be a separate API and **not** a shim that uses leader-{get, set} before 2.7 and peer app relation data after 2.7 with an upgrade path (in other words, we will not do anything behind a user's back).